Automattic/woocommerce-payments
# Use selling locations in international fraud protection filters

Fixes Automattic/woocommerce-payments-server#3148

#### Changes proposed in this Pull Request

This PR updates the international billing and ip address filters to use WC Core selling location(s) input of the merchant to match the IP and billing countries against. Before this PR, it was using the Stripe account country, which doesn't fulfill the requirements being written on the UI design. The title suggests that these filters act on merchant's allowed/blocked country definitions, so this change was a requirement for that.

**Note:** You can test this with [the server PR](https://github.com/Automattic/woocommerce-payments-server/pull/3128) to see if the rule engine works with the new settings.









#### Testing instructions





* Checkout this branch.
* Open two tabs; 
    * one with http://localhost:8082/wp-admin/admin.php?page=wc-settings to be able to change the selling locations setting.
    * the second with http://localhost:8082/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Ffraud-protection to be able to save those rulesets.
Don't forget to replace the domains if you're not using the docker setup.

_Testing "all" countries allowed_
* On the WC settings page, click the _Selling location(s)_ select box, and select _Sell to all countries_ option, and save.
* Refresh the Fraud protection advanced settings page, and see that the International IP Address and International Billing Address filters show a notice that they won't be effective, because they are selling to all countries. 
* Enable international IP and billing address filters.
* Click "Save changes"
* Open the client DB and check that you have a record named `_transient_wcpay_fraud_protection_settings` containing this as a part of it: 

```json
[
    { 
        "key": "international_ip_address", 
        "outcome": "review", 
        "check": {
            "key": "ip_country",
            "operator": "in",
            "value": "",
        }
    },
    { 
        "key": "international_billing_address", 
        "outcome": "review", 
        "check": {
            "key": "billing_country",
            "operator": "in",
            "value": "",
        }
    }
]
```

_Testing allowed countries_
* On the WC settings page, click the _Selling location(s)_ select box, and select _Sell to specific countries_ option, add "United States" and "Canada" to that list, and save.
* Refresh the Fraud protection advanced settings page, and see that the International IP Address and International Billing Address filters show a notice that they will block orders when they don't originate from US or Canada. 
* Enable international IP and billing address filters.
* Click "Save changes"
* Open the client DB and check that you have a record named `_transient_wcpay_fraud_protection_settings` containing this as a part of it: 

```json
[
    { 
        "key": "international_ip_address", 
        "outcome": "review", 
        "check": {
            "key": "ip_country",
            "operator": "not_in",
            "value": "US|CA",
        }
    },
    { 
        "key": "international_billing_address", 
        "outcome": "review", 
        "check": {
            "key": "billing_country",
            "operator": "not_in",
            "value": "US|CA",
        }
    }
]
```

_Testing blocked countries_
* On the WC settings page, click the _Selling location(s)_ select box, and select _Sell to all countries, except for_ option, add "United States" and "Canada" to that list, and save.
* Refresh the Fraud protection advanced settings page, and see that the International IP Address and International Billing Address filters show a notice that they will allow all orders except the ones originating from US or Canada. 
* Enable international IP and billing address filters.
* Click "Save changes"
* Open the client DB and check that you have a record named `_transient_wcpay_fraud_protection_settings` containing this as a part of it: 

```json
[
    { 
        "key": "international_ip_address", 
        "outcome": "review", 
        "check": {
            "key": "ip_country",
            "operator": "in",
            "value": "US|CA",
        }
    },
    { 
        "key": "international_billing_address", 
        "outcome": "review", 
        "check": {
            "key": "billing_country",
            "operator": "in",
            "value": "US|CA",
        }
    }
]
```

-------------------


**Post merge**
